internal/release/promote/promotion.go

```diff
@@ -131,20 +133,23 @@ func (p *Promotion) Promote(opts Opts) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("selecting releases to promote: %w", err)
 	}
-	if !list.HasAnyPromotableReleases() {
-		p.promptProvider.PrintNoPromotableReleasesFound(opts.ReleasesFiltered, opts.SourceEnv, opts.TargetEnv)
-		return "", nil
-	}
 
 	if !opts.NoPrompt {
-		if err := p.preview(list); err != nil {
+		if err := p.preview(selectedList); err != nil {
 			return "", fmt.Errorf("previewing: %w", err)
 		}
 	}
 
+	invalidList := selectedList.GetNonPromotableReleases(opts.SourceEnv, opts.TargetEnv)
+	if len(invalidList) != 0 {
+		invalid := strings.Join(invalidList, ", ")
```

Comment:
This is not really a comment on your PR because the codebase is like this everywhere, but we should stop printing error messages and returning errors.

**The error should represent the message and then be handled and printed later.**

In this case this actually introduces a very little 🐛 . Since the only way to get here is if the err is nil, and so we would print the error message but then exit successfully.

This means that in CI Workflows build promotion could fail, but the workflow would pass.